components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -152,22 +152,56 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
                 destination.confirmTID(tid);
 
             } else if (uri.getScheme().equals(SAPConstants.SAP_BAPI_PROTOCOL_NAME)) {
+                String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
                 try {
                     OMElement payLoad,body;
                     body = messageContext.getEnvelope().getBody();
                     payLoad = body.getFirstChildWithName(new QName(RFCConstants.BAPIRFC));
-                    log.info("Received RFC/Meta DATA: " + payLoad);
+                    if (log.isDebugEnabled()){
+                        log.debug("Received RFC/Meta DATA: " + payLoad);
+                    }
                     String rfcFunctionName = RFCMetaDataParser.getBAPIRFCFucntionName(payLoad);
-                    log.info("Looking up the BAPI/RFC function: " + rfcFunctionName + ". In the " +
-                             "meta data repository");
-
-                    JCoFunction function = getRFCfunction(destination, rfcFunctionName);
-                    RFCMetaDataParser.processMetaDataDocument(payLoad, function);
-                    String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
-                    String responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                    if (log.isDebugEnabled()){
+                        log.debug("Looking up the BAPI/RFC function: " + rfcFunctionName + ". In the " +
+                                "meta data repository");
+                    }
+                    String responseXML;
+                    if (isTransaction(messageContext)) {
```

Comment:
Shall we explain, what we are doing with this code segment for better readability??

Reply:
+1. Fixed.